homeassistant/components/hue/config_flow.py

```diff
@@ -194,26 +195,30 @@ async def async_step_ssdp(self, discovery_info: ssdp.SsdpServiceInfo) -> FlowRes
         """
         # Filter out non-Hue bridges #1
         if (
-            discovery_info.get(ssdp.ATTR_UPNP_MANUFACTURER_URL)
+            discovery_info.upnp.get(ssdp.ATTR_UPNP_MANUFACTURER_URL)
             not in HUE_MANUFACTURERURL
         ):
             return self.async_abort(reason="not_hue_bridge")
 
         # Filter out non-Hue bridges #2
         if any(
-            name in discovery_info.get(ssdp.ATTR_UPNP_FRIENDLY_NAME, "")
+            name in discovery_info.upnp.get(ssdp.ATTR_UPNP_FRIENDLY_NAME, "")
             for name in HUE_IGNORED_BRIDGE_NAMES
         ):
             return self.async_abort(reason="not_hue_bridge")
 
         if (
-            not discovery_info.get(ssdp.ATTR_SSDP_LOCATION)
-            or ssdp.ATTR_UPNP_SERIAL not in discovery_info
+            not discovery_info.ssdp_location
+            or ssdp.ATTR_UPNP_SERIAL not in discovery_info.upnp
         ):
             return self.async_abort(reason="not_hue_bridge")
 
-        host = urlparse(discovery_info[ssdp.ATTR_SSDP_LOCATION]).hostname
-        bridge = await self._get_bridge(host, discovery_info[ssdp.ATTR_UPNP_SERIAL])
+        host = urlparse(discovery_info.ssdp_location).hostname
+        if TYPE_CHECKING:
+            assert host is not None
```

Comment:
Should we abort instead?

Reply:
I have adjusted the host/location check to this:
```python
    if not host:
            return self.async_abort(reason="not_hue_bridge")
```